apps/vc-api/src/vc-api/exchanges/dtos/presentation-definition.dto.ts

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright 2021, 2022 Energy Web Foundation
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import { ApiProperty } from '@nestjs/swagger';
+import { IsArray, IsString } from 'class-validator';
+
+export class PresentationDefinitionDto {
+  @IsString()
+  @ApiProperty()
+  id: string;
+
+  @IsArray()
+  @ApiProperty({ isArray: true })
+  // TODO: consider defining DTO and validations
+  input_descriptors: Record<any, unknown>[];
```

Comment:
> Maybe a DTO for input descriptor needs to be defined so that the items type can be specified?

If this is the case, just doing a `Record<string, unknown>` for the InputDescriptorDTO would be enough, I think

Reply:
Something is incorrect here. This is Swagger is generated:
```json
"PresentationDefinitionDto": {
        "type": "object",
        "properties": {
          "id": { "type": "string" },
          "input_descriptors": { "type": "array", "items": { "type": "array" } }
        },
        "required": ["id", "input_descriptors"]
      }
```

I am checking how to solve this.

Reply:
I see what is wrong here. It should be:
```typescript
  @ApiProperty({ type: 'object', isArray: true })
```